### PR TITLE
[MM-39637] - Migrate from gorp to sqlx in store/sqlstore/shared_channel_store.go

### PR DESCRIPTION
**Pull request by penthaapatel**
#### Summary

Migrate from `gorp` to `sqlx` in `store/sqlstore/shared_channel_store.go`
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18875
JIRA: https://mattermost.atlassian.net/browse/MM-39637
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/shared_channel_store.go [hunk 1]
```diff
@@ -336,11 +371,31 @@ func (s SqlSharedChannelStore) UpdateRemote(remote *model.SharedChannelRemote) (
...
+		Set("Id", remote.Id).
+		Set("ChannelId", remote.ChannelId).
+		Set("CreatorId", remote.CreatorId).
+		Set("CreateAt", remote.CreateAt).
+		Set("UpdateAt", remote.UpdateAt).
+		Set("IsInviteAccepted", remote.IsInviteAccepted).
+		Set("IsInviteConfirmed", remote.IsInviteConfirmed).
+		Set("LastPostUpdateAt", remote.LastPostUpdateAt).
+		Set("LastPostId", remote.LastPostId).
+		Where(sq.Eq{"RemoteId": remote.RemoteId}).
```
Comment (agnivade) [inline store/sqlstore/shared_channel_store.go:384]: The where clause should have `id` and `channelid` as those are the primary key.

But even more importantly, this method is not used anywhere. @wiggin77 - wondering if we can remove this method? (in a separate PR of course)

File: store/sqlstore/shared_channel_store.go [hunk 2]
```diff
@@ -721,37 +789,23 @@ func (s SqlSharedChannelStore) UpsertAttachment(attachment *model.SharedChannelA
...
-			`INSERT INTO
-				SharedChannelAttachments
-				(Id, FileId, RemoteId, CreateAt, LastSyncAt)
-			VALUES
-				(:Id, :FileId, :RemoteId, :CreateAt, :LastSyncAt)
-			ON DUPLICATE KEY UPDATE
-				LastSyncAt = :LastSyncAt`, params); err != nil {
-			return "", err
-		}
+		query = query.SuffixExpr(sq.Expr("ON DUPLICATE KEY UPDATE LastSyncAt = ?", attachment.LastSyncAt))
```
Comment (agnivade) [inline store/sqlstore/shared_channel_store.go:798]: Nice!

**Review by agnivade — COMMENTED** [review]
Great work! Just one small comment.

**Review by agnivade — APPROVED** [review]
Fantastic!

**Review by isacikgoz — APPROVED** [review]
LGTM, nice work 💪

**Pull request merged**